pinot-core/src/test/java/org/apache/pinot/queries/NullEnabledQueriesTest.java

```diff
@@ -156,131 +179,186 @@ private void setUp(TableConfig tableConfig, DataType dataType)
     noDictionaryColumns.add(COLUMN_NAME);
     TableConfig tableConfig = new TableConfigBuilder(TableType.OFFLINE)
         .setTableName(RAW_TABLE_NAME)
         .setNoDictionaryColumns(noDictionaryColumns)
         .build();
     setUp(tableConfig, columnDataType.toDataType());
-    testQueries(baseValue, columnDataType, generateNulls);
+    testQueriesGeneric(columnDataType, generateNulls);
   }
 
-  @Test(priority = 4)
-  public void testQueriesWithDictFloatColumnNoNullValues()
-          throws Exception {
-    ColumnDataType columnDataType = ColumnDataType.FLOAT;
-    float baseValue = RANDOM.nextFloat();
-    boolean generateNulls = false;
-    createRecords(baseValue, generateNulls);
-    TableConfig tableConfig = new TableConfigBuilder(TableType.OFFLINE)
-            .setTableName(RAW_TABLE_NAME)
-            .build();
-    setUp(tableConfig, columnDataType.toDataType());
-    testQueries(baseValue, columnDataType, generateNulls);
+  @DataProvider(name = "numberColumnTypeProvider")
+  public Object[][] provideNumberColumnTypeAndRandomVal() {
+    return new Object[][]{
+        new Object[]{ColumnDataType.FLOAT, RANDOM.nextFloat(), true},
+        new Object[]{ColumnDataType.DOUBLE, RANDOM.nextDouble(), true},
+        new Object[]{ColumnDataType.FLOAT, RANDOM.nextFloat(), false},
+        new Object[]{ColumnDataType.DOUBLE, RANDOM.nextDouble(), false},
+    };
   }
 
-  @Test(priority = 5)
-  public void testQueriesWithNoDictFloatColumnNoNullValues()
-          throws Exception {
-    ColumnDataType columnDataType = ColumnDataType.FLOAT;
-    float baseValue = RANDOM.nextFloat();
-    boolean generateNulls = false;
-    createRecords(baseValue, generateNulls);
-    List<String> noDictionaryColumns = new ArrayList<String>();
-    noDictionaryColumns.add(COLUMN_NAME);
-    TableConfig tableConfig = new TableConfigBuilder(TableType.OFFLINE)
-            .setTableName(RAW_TABLE_NAME)
-            .setNoDictionaryColumns(noDictionaryColumns)
-            .build();
-    setUp(tableConfig, columnDataType.toDataType());
-    testQueries(baseValue, columnDataType, generateNulls);
+  @DataProvider(name = "otherColumnTypeProvider")
+  public Object[][] provideOtherColumnTypeAndRandomVal() {
+    return new Object[][]{
+        new Object[]{ColumnDataType.INT, RANDOM.nextInt(), true},
+        new Object[]{ColumnDataType.LONG, RANDOM.nextInt(), true},
+        new Object[]{ColumnDataType.BOOLEAN, RANDOM.nextInt(2), true},
+        new Object[]{ColumnDataType.STRING, RANDOM.nextInt(), true},
+        new Object[]{ColumnDataType.TIMESTAMP, RANDOM.nextInt(), true},
+        new Object[]{ColumnDataType.INT, RANDOM.nextInt(), false},
+        new Object[]{ColumnDataType.LONG, RANDOM.nextInt(), false},
+        new Object[]{ColumnDataType.BOOLEAN, RANDOM.nextInt(2), false},
+        new Object[]{ColumnDataType.STRING, RANDOM.nextInt(), false},
+        new Object[]{ColumnDataType.TIMESTAMP, RANDOM.nextInt(), false},
```

Comment:
this list is not at all a full coverage, but good to add STRING BOOLEAN and TIMESTAMP as representatives. can add more later

Reply:
@walterddr NullHandling support was tested against all data types. Please check the following files:

NullEnabledQueriesTest (Dict and non-dict Float and Double types).
BooleanNullEnabledQueriesTest (Dict and non-dict Boolean type).
BigDecimalQueriesTest (Dict and non-dict type).
AllNullQueriesTest (Dict and non-dict Long, Float, Double, Int, String and BigDecimal types).

Reply:
Ah I see thanks for the info. I wasn't aware of the other tests